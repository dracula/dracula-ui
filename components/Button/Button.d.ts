import React, { HTMLAttributes } from 'react';
import { colors as backgroundColors } from '../../base/colors';
export declare const buttonVariants: {
    normal: string;
    outline: string;
    ghost: string;
};
export declare const buttonSizes: {
    large: string;
    medium: string;
    small: string;
    xsmall: string;
};
/** Button Props */
export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
    /** A Dracula UI theme color for the Button. */
    color?: keyof typeof backgroundColors;
    /**
     * Controls the size of the button based on pre-configured Dracula UI sizes.
     * Options: `xsmall`, `small`, `medium`, `large`
     */
    size?: keyof typeof buttonSizes;
    /**
     * Controls the Button style:
     * `normal` -> Default solid Button with background color.
     * `outline` -> A subtle variation of the Button component with a softer background color that highlights the action text.
     * `ghost` -> A less prominent variation of the Button component that highlights hover interactions.
     */
    variant?: keyof typeof buttonVariants;
    /**
     * Controls the Button state. Mirrors the HTMLButtonElement `disabled` property.
     */
    disabled?: boolean;
    as?: 'button' | 'a' | 'input';
}
/**
 * The Button component triggers actions, behaviors, or events based
 * on user input.
 */
export declare const Button: React.FC<ButtonProps>;
