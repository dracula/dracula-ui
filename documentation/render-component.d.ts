import React from 'react';
export declare type ComponentExample = {
    html: string;
    docs: string;
    title: string;
    react?: any;
    screenshot?: string;
    svgData?: string;
};
export declare type SnapshotBuilder = {
    comp: () => React.ReactElement;
    docs: string;
    title: string;
};
export declare function snapshot(title: string, comp: () => React.ReactElement, docs?: string): SnapshotBuilder;
export declare function renderSnapshot(title: string, comp: () => React.ReactElement, docs?: string): ComponentExample;
