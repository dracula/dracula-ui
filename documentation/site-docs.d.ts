import { SnapshotBuilder } from './render-component';
import React from 'react';
interface Documentation {
    basic: () => SnapshotBuilder;
    variations: () => Array<SnapshotBuilder> | SnapshotBuilder;
}
export declare function docs<T>(componentClass: React.FunctionComponent<T> | string, documentation: Documentation): void;
export {};
