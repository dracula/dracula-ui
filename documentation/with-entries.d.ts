import { SnapshotBuilder } from './render-component';
export declare function withEntries<T>(map: T, fn: (name: keyof T, value: T[keyof T]) => SnapshotBuilder): SnapshotBuilder[];
