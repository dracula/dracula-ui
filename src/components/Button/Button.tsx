import React, { HTMLAttributes } from 'react'
import { colors as backgroundColors } from '../../base/colors'
import cx from 'classnames'
import { textColors } from '../../components/Typography/Text'

export const buttonVariants = {
  normal: 'drac-btn',
  outline: 'drac-btn-outline',
  ghost: 'drac-btn-ghost'
}

export const buttonSizes = {
  large: 'drac-btn-lg',
  medium: 'drac-btn',
  small: 'drac-btn-sm',
  xsmall: 'drac-btn-xs'
}

/** Button Props */
export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
  /** A Dracula UI theme color for the Button. */
  color?: keyof typeof backgroundColors

  /**
   * Controls the size of the button based on pre-configured Dracula UI sizes.
   * Options: `xsmall`, `small`, `medium`, `large`
   */
  size?: keyof typeof buttonSizes

  /**
   * Controls the Button style:
   * `normal` -> Default solid Button with background color.
   * `outline` -> A subtle variation of the Button component with a softer background color that highlights the action text.
   * `ghost` -> A less prominent variation of the Button component that highlights hover interactions.
   */
  variant?: keyof typeof buttonVariants

  /**
   * Controls the Button state. Mirrors the HTMLButtonElement `disabled` property.
   */
  disabled?: boolean
}

/**
 * The Button component triggers actions, behaviors, or events based
 * on user input.
 */
export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  const isOutline = props.variant === 'outline'
  const isGhost = props.variant === 'ghost'
  const overrideTextColor = isOutline || isGhost

  const textColorClass = overrideTextColor
    ? textColors[props.color ?? 'green']
    : undefined

  let backgroundClass = backgroundColors[props.color ?? 'green']
  if (isGhost) {
    backgroundClass = `${backgroundClass}-transparent`
  }

  const classes = cx(
    'drac-btn',
    backgroundClass,
    buttonVariants[props.variant ?? 'normal'],
    buttonSizes[props.size ?? 'medium'],
    textColorClass
  )

  return (
    <button className={classes} {...props}>
      {props.children}
    </button>
  )
}

Button.displayName = 'Button'
