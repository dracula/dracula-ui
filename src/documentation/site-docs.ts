import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '../story-helpers/render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from '../story-helpers/pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

jest.setTimeout(10000)

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name
  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeEach(async () => {
      browser = await puppeteer.launch({ headless: true, devtools: true })
    })

    afterEach(async () => {
      await browser.close()
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      const [screenshot, svg] = await componentScreenshot(
        browser,
        snapshot,
        null,
        name
      )

      snapshot.screenshot = screenshot
      snapshot.svgData = svg

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          variation,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
