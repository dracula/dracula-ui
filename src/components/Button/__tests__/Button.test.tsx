import React from 'react'
import { Button } from '../Button'
import {
  buildSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import { siteDocs } from '@/documentation/site-docs'
import {
  ButtonVariants,
  ButtonSizes,
  ButtonColors
} from '@/components/Button/__stories__/Button.stories'

siteDocs(Button, {
  basicUsage() {
    return buildSnapshot('Usage', () => <Button color="purple">Dracula</Button>)
  },
  variations() {
    return [
      buildSnapshot('Variants', ButtonVariants),
      buildSnapshot('Sizes', ButtonSizes),
      buildSnapshot('Colors', ButtonColors)
    ] as SnapshotBuilder[]
  }
})
