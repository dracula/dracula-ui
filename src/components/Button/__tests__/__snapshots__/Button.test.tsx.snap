// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Button Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<button class=\\"drac-btn drac-bg-purple drac-btn drac-btn\\" color=\\"purple\\">
  Dracula
</button>
",
  "react": "<Button color=\\"purple\\">Dracula</Button>;
",
  "title": "Usage",
}
`;

exports[`Site: Button Colors 1`] = `
Object {
  "docs": "",
  "html": "<div class=\\"drac-box\\">
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-white drac-btn drac-btn\\"
      color=\\"white\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-white drac-btn-outline drac-btn drac-text-white\\"
      color=\\"white\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-white-transparent drac-btn-ghost drac-btn drac-text-white\\"
      color=\\"white\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-cyan drac-btn drac-btn\\"
      color=\\"cyan\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-cyan drac-btn-outline drac-btn drac-text-cyan\\"
      color=\\"cyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-cyan-transparent drac-btn-ghost drac-btn drac-text-cyan\\"
      color=\\"cyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-green drac-btn drac-btn\\"
      color=\\"green\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-green drac-btn-outline drac-btn drac-text-green\\"
      color=\\"green\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-green-transparent drac-btn-ghost drac-btn drac-text-green\\"
      color=\\"green\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-orange drac-btn drac-btn\\"
      color=\\"orange\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-orange drac-btn-outline drac-btn drac-text-orange\\"
      color=\\"orange\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-orange-transparent drac-btn-ghost drac-btn drac-text-orange\\"
      color=\\"orange\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-pink drac-btn drac-btn\\"
      color=\\"pink\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-pink drac-btn-outline drac-btn drac-text-pink\\"
      color=\\"pink\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-pink-transparent drac-btn-ghost drac-btn drac-text-pink\\"
      color=\\"pink\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple drac-btn drac-btn\\"
      color=\\"purple\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-purple drac-btn-outline drac-btn drac-text-purple\\"
      color=\\"purple\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-purple-transparent drac-btn-ghost drac-btn drac-text-purple\\"
      color=\\"purple\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-red drac-btn drac-btn\\"
      color=\\"red\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-red drac-btn-outline drac-btn drac-text-red\\"
      color=\\"red\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-red-transparent drac-btn-ghost drac-btn drac-text-red\\"
      color=\\"red\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-yellow drac-btn drac-btn\\"
      color=\\"yellow\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-yellow drac-btn-outline drac-btn drac-text-yellow\\"
      color=\\"yellow\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-yellow-transparent drac-btn-ghost drac-btn drac-text-yellow\\"
      color=\\"yellow\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple-cyan drac-btn drac-btn\\"
      color=\\"purpleCyan\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-purple-cyan drac-btn-outline drac-btn drac-text-purple-cyan\\"
      color=\\"purpleCyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-purple-cyan-transparent drac-btn-ghost drac-btn drac-text-purple-cyan\\"
      color=\\"purpleCyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-yellow-pink drac-btn drac-btn\\"
      color=\\"yellowPink\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-yellow-pink drac-btn-outline drac-btn drac-text-yellow-pink\\"
      color=\\"yellowPink\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-yellow-pink-transparent drac-btn-ghost drac-btn drac-text-yellow-pink\\"
      color=\\"yellowPink\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-cyan-green drac-btn drac-btn\\"
      color=\\"cyanGreen\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-cyan-green drac-btn-outline drac-btn drac-text-cyan-green\\"
      color=\\"cyanGreen\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-cyan-green-transparent drac-btn-ghost drac-btn drac-text-cyan-green\\"
      color=\\"cyanGreen\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-pink-purple drac-btn drac-btn\\"
      color=\\"pinkPurple\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-pink-purple drac-btn-outline drac-btn drac-text-pink-purple\\"
      color=\\"pinkPurple\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-pink-purple-transparent drac-btn-ghost drac-btn drac-text-pink-purple\\"
      color=\\"pinkPurple\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-animated drac-btn drac-btn\\"
      color=\\"animated\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-animated drac-btn-outline drac-btn drac-text-animated\\"
      color=\\"animated\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-animated-transparent drac-btn-ghost drac-btn drac-text-animated\\"
      color=\\"animated\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
</div>
",
  "react": "<Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"white\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"white\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"white\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"cyan\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"cyan\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"cyan\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"green\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"green\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"green\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"orange\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"orange\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"orange\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"pink\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"pink\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"pink\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"purple\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"purple\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"purple\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"red\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"red\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"red\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"yellow\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"yellow\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"yellow\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"purpleCyan\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"purpleCyan\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"purpleCyan\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"yellowPink\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"yellowPink\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"yellowPink\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"cyanGreen\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"cyanGreen\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"cyanGreen\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"pinkPurple\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"pinkPurple\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"pinkPurple\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"animated\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"animated\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"animated\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Button Sizes 1`] = `
Object {
  "docs": "",
  "html": "<div class=\\"drac-box\\">
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple drac-btn drac-btn-lg\\" color=\\"purple\\">
      large
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple drac-btn drac-btn\\" color=\\"purple\\">
      medium
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple drac-btn drac-btn-sm\\" color=\\"purple\\">
      small
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple drac-btn drac-btn-xs\\" color=\\"purple\\">
      xsmall
    </button>
  </div>
</div>
",
  "react": "<Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"large\\">
      large
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"medium\\">
      medium
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"small\\">
      small
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"xsmall\\">
      xsmall
    </Button>
  </Box>
</Box>;
",
  "title": "Sizes",
}
`;

exports[`Site: Button Variants 1`] = `
Object {
  "docs": "",
  "html": "<div class=\\"drac-box\\">
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple drac-btn drac-btn\\"
      variant=\\"normal\\"
      color=\\"purple\\"
    >
      normal
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple drac-btn-outline drac-btn drac-text-purple\\"
      variant=\\"outline\\"
      color=\\"purple\\"
    >
      outline
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple-transparent drac-btn-ghost drac-btn drac-text-purple\\"
      variant=\\"ghost\\"
      color=\\"purple\\"
    >
      ghost
    </button>
  </div>
</div>
",
  "react": "<Box>
  <Box style={{ marginBottom: 10 }}>
    <Button variant=\\"normal\\" color=\\"purple\\">
      normal
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button variant=\\"outline\\" color=\\"purple\\">
      outline
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button variant=\\"ghost\\" color=\\"purple\\">
      ghost
    </Button>
  </Box>
</Box>;
",
  "title": "Variants",
}
`;
